vendor/processors/stripe_processor.py

```diff
@@ -199,6 +269,203 @@ def build_subscription(self, subscription, payment_method_id):
+            stripe_product_id = meta.get('stripe', {}).get('product_id', {})
+            stripe_price_id = meta.get('stripe', {}).get('price_id', {})
+            stripe_coupon_id = meta.get('stripe', {}).get('coupon_id', {})
+            product_data = self.build_product(offer)
+            price_data = self.build_price(offer, offer.current_price_object())
+            coupon_data = self.build_coupon(offer, offer.current_price_object())
+
+            if not stripe_product_id:
+                product = self.stripe_create_object(self.stripe.Product, product_data)
+
+                if meta.get('stripe', None):
+                    meta['stripe']['product_id'] = product['id']
+                else:
+                    meta['stripe'] = {'product_id': product['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_product = self.stripe_update_object(self.stripe.Product, stripe_product_id, product_data)
+
+                if existing_product:
+                    if meta.get('stripe', None):
+                        meta['stripe']['product_id'] = existing_product['id']
+                    else:
+                        meta['stripe'] = {'product_id': existing_product['id']}
+
+            # Check if meta has this offer stripe price and add if not
+            if not stripe_price_id:
+                price = self.stripe_create_object(self.stripe.Price, price_data)
+
+                if meta.get('stripe', None):
+                    meta['stripe']['price_id'] = price['id']
+                else:
+                    meta['stripe'] = {'price_id': price['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_price = self.stripe_update_object(self.stripe.Price, stripe_price_id, price_data)
+
+                if existing_price:
+                    if meta.get('stripe', None):
+                        meta['stripe']['price_id'] = existing_price['id']
+                    else:
+                        meta['stripe'] = {'price_id': existing_price['id']}
+
+            # Check if meta has this offer stripe coupon and add if not
+            if not stripe_coupon_id:
+                coupon = self.stripe_create_object(self.stripe.Coupon, coupon_data)
+
+                if meta.get('stripe', None):
+                    meta['stripe']['coupon_id'] = coupon['id']
+                else:
+                    meta['stripe'] = {'coupon_id': coupon['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_coupon = self.stripe_update_object(self.stripe.Coupon, stripe_coupon_id, coupon_data)
+
+                if existing_coupon:
+                    if meta.get('stripe', None):
+                        meta['stripe']['coupon_id'] = existing_coupon['id']
+                    else:
+                        meta['stripe'] = {'coupon_id': existing_coupon['id']}
+
+            offer.save()
```

Comment:
I would follow the same logic as with the customers.


```
1. Get Site Offers.
2. Get Offers to create on Stripe.
3. Get Offers to update on Stripe.
4. Create Offers with their price and coupon
5. Update Offers.
5.1 Check if the offer already has an price on Stripe to update or create.
5.2 Check if the offer has a trial period or discrount. 
5.2.1 Check if it has it on Stripe to create, update or delete.
```

Reply:
Updated